### PR TITLE
framework/task_manager: Modify Task Manager to wait unicast-reply unt…

### PR DESCRIPTION
**Pull request by jeongchanKim**
…il timeout expired

If no reply until timeout expired when unicast, task manager goes to next step

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [44a69d4500a2e2f7704b0515958103c30a7e2404] -

**Comment by seinfra** [comment]
Target : [44a69d4500a2e2f7704b0515958103c30a7e2404] -

**Comment by seinfra** [comment]
Target : [8d26b026032cb45523259c160871cea11893ba19] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [8d26b026032cb45523259c160871cea11893ba19] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [1b68f067aebf87125c0024815c2fa749b5248a27] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [1b68f067aebf87125c0024815c2fa749b5248a27] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [9a0a5a74219bdbfca9ae6b7b12f67009f5afd344] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [36e945e7398fd576d056017f74b3f5351a9a36b2] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [36e945e7398fd576d056017f74b3f5351a9a36b2] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [fccf1a93fcfdc8b2d86d8c0ce23b10fa6a3d45d2] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [fccf1a93fcfdc8b2d86d8c0ce23b10fa6a3d45d2] - Code Rule Check OK.

File: framework/src/task_manager/task_manager_core.c [hunk 1]
```diff
@@ -452,14 +454,15 @@ static int taskmgr_unicast_async(int handle, int caller_pid, void *data)
 	return OK;
 }
 
-static int taskmgr_unicast_sync(int handle, int caller_pid, tm_unicast_internal_msg_t *data, tm_unicast_msg_t *response_msg)
+static int taskmgr_unicast_sync(int handle, int caller_pid, tm_unicast_internal_msg_t *data, tm_unicast_msg_t *response_msg, int timeout)
```
[the remark this answers is not in the document]
Comment (jeongchanKim) [reply framework/src/task_manager/task_manager_core.c]: usually -1 is used to present infinite

**Comment by seinfra** [comment]
Target : [5d3e3e4b9e14b7efcb1a09b2a7adb639d7ee8e30] -

**Comment by seinfra** [comment]
Target : [5d3e3e4b9e14b7efcb1a09b2a7adb639d7ee8e30] -

**Comment by seinfra** [comment]
Target : [050038dd6b5c1b93e0a77e259c1cd78d7e4bdf07] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [050038dd6b5c1b93e0a77e259c1cd78d7e4bdf07] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [6f577da5a4b2918179aec2bed14a6403b7031336] -

**Comment by seinfra** [comment]
Target : [6f577da5a4b2918179aec2bed14a6403b7031336] -

File: framework/src/task_manager/Kconfig [hunk 2]
```diff
@@ -27,9 +27,19 @@ config TASK_MANAGER_MAX_MSG
 	---help---
 		Task Manager can receive under this value.
 
+
```
Comment (sunghan-chang) [inline framework/src/task_manager/Kconfig]: Let's remove a meanless change.

File: framework/include/task_manager/task_manager.h [hunk 3]
```diff
@@ -269,8 +270,11 @@ int task_manager_restart(int handle, int timeout);
  * @param[in] timeout returnable flag. It can be one of the below.\n
  *			TM_NO_RESPONSE : Ignore the response of request from task manager\n
  *			TM_RESPONSE_WAIT_INF : Blocked until get the response from task manager\n
- *			integer value : Specifies an upper limit on the time for which will block in milliseconds
- *            If reply is not NULL(=SYNC), timeout should not be set to TM_NO_RESPONSE.
+ *			integer value : Specifies an upper limit on the time for which will block in milliseconds\n
+ *            If reply is not NULL(=SYNC), timeout should not be set to TM_NO_RESPONSE.\n
+ *            When you enable CONFIG_TASK_MANAGER_UNICAST_REPLY_TIMEOUT with non-zero value, task manager waits the reply during\n
+ *            CONFIG_TASK_MANAGER_UNICAST_REPLY_TIMEOUT even if user calls this API with TM_RESPONSE_WAIT_INF.\n
+ *            When not set, task manager waits the reply forever, it can cause task manager hang.
```
Comment (sunghan-chang) [inline framework/include/task_manager/task_manager.h]: When it has zero value,

**Comment by seinfra** [comment]
Target : [07663ca877ce50ffd05409cab13e97cd1368e354] -

**Comment by seinfra** [comment]
Target : [07663ca877ce50ffd05409cab13e97cd1368e354] -

**Pull request merged**